File: Projects/Project_1/Fox.cpp
#include "Fox.h"

Fox::Fox(int  whenWasBorn, Coordinates coords, World* world) : Animal(3, 7, whenWasBorn, 1, 'F', coords, world, Organism::TypeOfOrganism::FOX, false) {

}

void Fox::action() {
	int x = this->GetCoordinates().GetX();
	int y = this->GetCoordinates().GetY();
	this->SetFreeDirections(moveRange);
	Direction d = this->randDirection();
	if (d == Direction::UP) {
		y -= moveRange;
	}
	else if (d == Direction::DOWN) {
		y += moveRange;
	}
	else if (d == Direction::LEFT) {
		x += moveRange;
	}
	else if (d == Direction::RIGHT) {
		x -= moveRange;
	}
	Coordinates c1 = this->GetCoordinates();
	this->SetCoorinates(x, y);
	world->GetBoard()->moveOrganism(this, c1);
}

void Fox::GetOrganismName() {
	cout << "Fox";
}

Fox::~Fox(){}

File: Projects/po_projekt2/organisms/animals/Fox.java
package po_projekt2.organisms.animals;

import java.awt.Color;

import po_projekt2.World;
import po_projekt2.organisms.Animal;
import po_projekt2.worldElements.Coordinates;
import po_projekt2.Organism.typeOfOrganism;

/**
 *
 * @author Bogna
 */
public class Fox extends Animal{
    
    public Fox(int  whenWasBorn, Coordinates coords, World world){
        super(3, 7, whenWasBorn, 1, new Color(255,140,0), "Fox",coords, world, typeOfOrganism.FOX, false);
    }
    
    @Override
    public void action() {
	int x = this.getCoords().getX();
	int y = this.getCoords().getY();
	this.setFreeDirections(moveRange);
	direction d = this.randDirection();
	if (d == direction.UP) {
            y -= moveRange;
	}
	else if (d == direction.DOWN) {
            y += moveRange;
	}
	else if (d == direction.LEFT) {
            x += moveRange;
	}
	else if (d == direction.RIGHT) {
            x -= moveRange;
	}
	Coordinates c1 = this.getCoords();
	this.setCoords(x, y);
	world.getBoard().moveOrganism(this, c1);
    }
    
//    @Override
//    public String getOrganismName() {
//	return "Fox ";
//    }
}
